fast64_internal/f3d/f3d_material.py

```diff
@@ -180,17 +169,52 @@ def update_draw_layer(self, context):
         set_output_node_groups(material)
 
 
+world_layer_defaults_attrs = {
+    "SM64": lambda world, layer: (
+        getattr(world, f"draw_layer_{layer}_cycle_1", ""),
+        getattr(world, f"draw_layer_{layer}_cycle_2", ""),
+    ),
+    "OOT": lambda world, layer: (
+        getattr(world.ootDefaultRenderModes, f"{layer.lower()}Cycle1", ""),
+        getattr(world.ootDefaultRenderModes, f"{layer.lower()}Cycle2", ""),
+    ),
+}
+
+
 def rendermode_preset_to_advanced(material: bpy.types.Material):
     """
     Set all individual controls for the rendermode from the preset rendermode.
     """
-    settings = material.f3d_mat.rdp_settings
+    scene = bpy.context.scene
+    f3d_mat = material.f3d_mat
+    settings = f3d_mat.rdp_settings
     f3d = get_F3D_GBI()
 
-    if settings.rendermode_advanced_enabled:
+    if settings.rendermode_advanced_enabled and settings.set_rendermode:
         # Already in advanced mode, don't overwrite this with the preset
         return
 
+    if settings.set_rendermode:
+        cycle_1, cycle_2 = settings.rendermode_preset_cycle_1, settings.rendermode_preset_cycle_2
+    else:
+        game_mode = scene.gameEditorMode
+        layer = getattr(f3d_mat.draw_layer, game_mode.lower(), None)
+        if layer is None:  # Game mode has no layer, don´t change anything
+            return
+
+        # If no world defaults for game mode, fallback on get_with_default default arg
+        cycle_1, cycle_2 = default_draw_layers.get(game_mode, {}).get(layer, ("", ""))
+        if scene.world is not None and game_mode in world_layer_defaults_attrs:
+            possible_cycle_1, possible_cycle_2 = world_layer_defaults_attrs[game_mode](scene.world, layer)
```

Comment:
this looks really weird to me, I think it would be better if ``world_layer_defaults_attrs`` was a real function